crates/floresta-wire/src/p2p_wire/node.rs

```diff
@@ -210,6 +214,25 @@ where
     ) -> Self {
         let (node_tx, node_rx) = unbounded_channel();
         let socks5 = config.proxy.map(Socks5StreamBuilder::new);
+
+        let fixed_peer = config.fixed_peer.as_ref().map(|address| {
+            let address =
+                Self::resolve_connect_host(&address, Self::get_port(config.network.into()));
+            match address {
+                Ok(address) => {
+                    info!(
+                        "Using {} as fixed peer",
+                        address.get_net_address().to_string()
+                    );
+                    address
+                }
+                Err(e) => {
+                    warn!("Invalid fixed peer address: {:?}", e);
+                    exit(1);
```

Comment:
The `fixed_peer` block calls `exit(1) `if an error occurs while resolving the address. This may abruptly terminate the application, which is generally not recommended for libraries or components that could be reused. Is this intencional?

Recomendation:
- Return a `Result<Self, Error>` from new to propagate errors upward, allowing the caller to decide how to handle them.
- Log the error and use a None default, continuing execution if the fixed peer is optional

```rust
let fixed_peer = match config.fixed_peer.as_ref().map(|address| {
    Self::resolve_connect_host(&address, Self::get_port(config.network.into()))
}) {
    Some(Ok(address)) => {
        info!("Using {} as fixed peer", address.get_net_address().to_string());
        Some(address)
    }
    Some(Err(e)) => {
        warn!("Invalid fixed peer address: {:?}", e);
        None
    }
    None => None,
};
```